File: reimi/frontend/reimi_app/lib/domain/value_objects/weather_type.dart
import 'package:freezed_annotation/freezed_annotation.dart';

enum WeatherType {
  @JsonValue('CLEAR_STARS')
  clearStars, // 星がはっきり
  @JsonValue('FAINT_STARS')
  faintStars, // 星がうっすら
  @JsonValue('NO_STARS')
  noStars, // 星が見えない
  @JsonValue('DRIZZLE')
  drizzle, // ポツポツ
  @JsonValue('LIGHT_RAIN')
  lightRain, // パラパラ
  @JsonValue('RAIN')
  rain, // サー
  @JsonValue('HEAVY_RAIN')
  heavyRain, // サーサー
}

File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/report/value/WeatherType.java
package com.reimi.reimi_app.domain.model.report.value;

public enum WeatherType {
    CLEAR_STARS("星がはっきり"),
    FAINT_STARS("星がうっすら"),
    NO_STARS("星が見えない"),
    DRIZZLE("ポツポツ"),
    LIGHT_RAIN ("パラパラ"),
    RAIN("サー"),
    HEAVY_RAIN("サーサー");

    private final String label;

    WeatherType(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}